### PR TITLE
Add validators

### PR DESCRIPTION
**Pull request by tkkuehn**
This adds attribute validators to `AfidPosition` and `AfidSet`, ensuring that we don't allow invalid AFIDs to be instantiated.

This obsoleted some of the existing checks in ext, so I removed them.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -14,6 +14,7 @@ exclude = ["afids_utils/tests"]
 python = ">=3.8, <3.12"
 attrs = "^23.1.0"
 numpy = "^1.24.3"
+more-itertools = "^10.1.0"
```
Comment (kaitj) [inline pyproject.toml:17]: I might have missed this, but I thought this was only used in testing. In which case this can be moved to the dev group.

**Review by kaitj — APPROVED** [review]
Looks good to me with the one comment on where the the dependency is added. One thought (doesn't need to be this PR), is it possible to make the validation optional? There were some discussion of potentially having what is currently considered an "invalid" afid set be passed in.

Other than the comment, this can be merged in.

File: pyproject.toml (hunk 1 above)
Comment (tkkuehn) [reply pyproject.toml:17]: Ah, good catch.

**Comment by tkkuehn** [comment]
> is it possible to make the validation optional? There were some discussion of potentially having what is currently considered an "invalid" afid set be passed in.

`with attrs.validators.disabled():` as a context manager will disable the validation (not a super convenient API for users of afids-utils, admittedly). To make that an option here will require a bit of discussion of how much of the `AfidSet` API needs a full set of AFIDs. I think it's reasonable to offer maybe an alternate class to allow partial AFID sets or invalid fiducials, just depends on what users want from that class.

**Pull request merged**